Index: src/games/strategy/net/ServerMessenger.java
===================================================================
--- src/games/strategy/net/ServerMessenger.java	(revision 2786)
+++ src/games/strategy/net/ServerMessenger.java	(working copy)
@@ -209,7 +209,10 @@
     
     public void messageReceived(MessageHeader msg,  SocketChannel channel)
     {
-
+        INode expectedReceive = m_channelToNode.get(channel);
+        if(!expectedReceive.equals(msg.getFrom())) {
+            throw new IllegalStateException("expected:" + expectedReceive + " not:" + msg.getFrom());
+        }
         if (msg.getFor() == null)
         {
             forwardBroadcast(msg);
